### PR TITLE
BaSP-TG-83: Get and Delete of Tasks

### PR DESCRIPTION
**Pull request by FiolAxel**
1. Define and test the actions.
2. Define and test the constants.
3. Define and test the reducer.
4. Define and test the thunks actions.

### CODE REVIEW

**Review by Facund0Fernandez — APPROVED** [review]
Well done!

File: src/components/Tasks/index.js [hunk 1]
```diff
@@ -1,94 +1,125 @@
...
     }
-  }, []);
+  }, [error]);
 
-  const search = (id) => {
-    setSearchQuery(id);
-    const result = untouchedData.filter((task) => task._id.includes(id));
-    setDataTable(result);
+  useEffect(() => {
+    setFilteredList(tasks.filter((item) => item._id.includes(searchQuery)));
```
Comment (franco-marini) [inline src/components/Tasks/index.js:43]: Could you add that the letters of the filter should be in the same case?

File: src/components/Tasks/index.js [hunk 2]
```diff
@@ -1,94 +1,125 @@
...
   return (
     <section className={styles.containerTask}>
-      <Modal modalTitle={'Delete Task'} isOpen={isOpen} handleClose={() => setIsOpen(!isOpen)}>
-        <p>Do you wish to eliminate the task?</p>
+      <Modal
+        modalTitle={'Tasks'}
+        isOpen={isOpen}
+        handleClose={() => {
+          openModal();
+        }}
```
Comment (franco-marini) [inline src/components/Tasks/index.js:78]: Do you think that is ok to use that function on the `handleClose`?

File: src/components/Tasks/index.js [hunk 3]
```diff
@@ -1,94 +1,125 @@
...
       </Modal>
       <h2 className={styles.h2Task}>Tasks</h2>
       <div className={styles.containerTaskBS}>
-        <Button text={'Add Task'} link={'/tasks/form'} />
-        <Search searchQuery={searchQuery} setSearchQuery={search} placeholder="Search by ID" />
+        <Button
+          text={'Add Task'}
+          handler={() => {
+            history.push('/tasks/form');
+          }}
```
Comment (franco-marini) [inline src/components/Tasks/index.js:107]: Review this function with one of the team members

**Review by franco-marini — CHANGES_REQUESTED** [review]
Nice work! Review some of the comments

**Review by Facund0Fernandez — APPROVED** [review]

**Review by franco-marini — APPROVED** [review]
🚀

**Review by vadalamati — APPROVED** [review]
Good job!

**Pull request merged**